### PR TITLE
rename interface{} to any

### PR DESCRIPTION
**Pull request by izturn**
Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

### CODE REVIEW

File: design/custom-request-timeout.md [hunk 1]
```diff
@@ -173,7 +173,7 @@ type RetryPolicy struct {
 
 type Route struct {
     Prefix          string
-    object          interface{}
+    object          any
```
Comment (skriss) [inline design/custom-request-timeout.md:176]: We can omit the design docs from this update, no need to touch them.

File: design/loadbalancer-hash-policy-design.md [hunk 2]
```diff
@@ -150,10 +150,10 @@ For example, to implement header hashing, the `headerName` field would be requir
 This option was not chosen as it would require all future configuration fields to fit into a `string` value, making the possibility for more complex data types more difficult to use and validate.
 Strongly typed structs also match better with other configuration patterns we have used in Contour.
 
-### Using `map[string]interface{}` for `HashOptions` Type
+### Using `map[string]any` for `HashOptions` Type
```
Comment (skriss) [inline design/loadbalancer-hash-policy-design.md:153]: can omit changes to this file

**Review by skriss — CHANGES_REQUESTED** [review]

File: design/loadbalancer-hash-policy-design.md [hunk 3]
```diff
@@ -152,8 +152,8 @@ Strongly typed structs also match better with other configuration patterns we ha
 
 ### Using `map[string]interface{}` for `HashOptions` Type
 This approach does have the benefit of not restricting hash options to string values if we choose to support a wide range of options in the future.
-This option was considered and not chosen because using `interface{}` as the value type of the options map out of the box does not work with the deep copy infrastructure we have.
-We would likely have to implement a type alias to `interface{}` and write some custom deep copy code to support this rather than rely on types that work with out of the box tooling.
+This option was considered and not chosen because using `any` as the value type of the options map out of the box does not work with the deep copy infrastructure we have.
+We would likely have to implement a type alias to `any` and write some custom deep copy code to support this rather than rely on types that work with out of the box tooling.
```
Comment (skriss) [inline design/loadbalancer-hash-policy-design.md:156]: please revert this change too

File: design/loadbalancer-hash-policy-design.md [hunk 4]
```diff
@@ -152,8 +152,8 @@ Strongly typed structs also match better with other configuration patterns we ha
 
 ### Using `map[string]interface{}` for `HashOptions` Type
 This approach does have the benefit of not restricting hash options to string values if we choose to support a wide range of options in the future.
-This option was considered and not chosen because using `interface{}` as the value type of the options map out of the box does not work with the deep copy infrastructure we have.
-We would likely have to implement a type alias to `interface{}` and write some custom deep copy code to support this rather than rely on types that work with out of the box tooling.
+This option was considered and not chosen because using `any` as the value type of the options map out of the box does not work with the deep copy infrastructure we have.
```
Comment (skriss) [inline design/loadbalancer-hash-policy-design.md:155]: please revert this change too

**Review by skriss — CHANGES_REQUESTED** [review]

File: design/loadbalancer-hash-policy-design.md (hunk 3 above)
Comment (izturn) [reply design/loadbalancer-hash-policy-design.md]: done

**Review by skriss — APPROVED** [review]

**Comment by skriss** [comment]
@izturn I think if we want this to be a standard, then we need to add a linter for this, otherwise folks are likely to add back usage of `interface{}` in the future. Can you add that to this PR?

**Comment by izturn** [comment]
ok, i will add a linter for it

**Comment by izturn** [comment]
@skriss  PTAL

**Review by skriss — APPROVED** [review]

**Pull request merged**